src/Packagist/Api/Client.php

```diff
@@ -76,7 +76,10 @@ public function search(string $query, array $filters = [], int $limit = 0): arra
                 $createResult = [$createResult];
             }
             $results = array_merge($results, $createResult);
-        } while (isset($response['next']) && (0 === $limit || $response['next'] <= $limit));
+            if (isset($response['next'])) {
+                parse_str(parse_url($response['next'], PHP_URL_QUERY), $parse);
+            }
+        } while (isset($response['next']) && (0 === $limit || $parse['page'] <= $limit));
```

Comment:
Do you need to check `isset($parse)`? It looks like it's defined inside the condition abover

Reply:
$parse is present along with $response['next'], so logically a check on latter is enough to ensure the existence of former